src/renderer/redux/actions/publish.js

```diff
@@ -24,6 +34,82 @@ export const doUpdatePublishForm = (publishFormValue: UpdatePublishFormData) =>
     data: { ...publishFormValue },
   });
 
+export const doResetThumbnailStatus = () => (dispatch: Dispatch): Action =>
+  fetch('https://spee.ch/api/channel/availability/@testing')
+    .then(() =>
+      dispatch({
+        type: ACTIONS.UPDATE_PUBLISH_FORM,
+        data: {
+          uploadThumbnailStatus: STATUSES.READY,
+          thumbnail: '',
+          nsfw: false,
+        },
+      })
+    )
+    .catch(() =>
+      dispatch({
+        type: ACTIONS.UPDATE_PUBLISH_FORM,
+        data: {
+          uploadThumbnailStatus: STATUSES.API_DOWN,
+          thumbnail: '',
+          nsfw: false,
+        },
+      })
+    );
+
+export const doUploadThumbnail = (filePath: string, nsfw: boolean) => (dispatch: Dispatch) => {
+  const thumbnail = fs.readFileSync(filePath);
+  const fileExt = path.extname(filePath);
+
+  const makeid = () => {
+    let text = '';
+    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+    for (let i = 0; i < 24; i += 1) text += possible.charAt(Math.floor(Math.random() * 62));
+    return text;
+  };
+
+  const uploadError = (error = '') =>
+    dispatch(
+      batchActions(
+        {
+          type: ACTIONS.UPDATE_PUBLISH_FORM,
+          data: { uploadThumbnailStatus: STATUSES.API_DOWN },
+        },
+        doAlertError(error)
+      )
+    );
+
+  dispatch({
+    type: ACTIONS.UPDATE_PUBLISH_FORM,
+    data: { uploadThumbnailStatus: STATUSES.IN_PROGRESS },
+  });
+
+  const data = new FormData();
+  const name = makeid();
+  const blob = new Blob([thumbnail], { type: `image/${fileExt.slice(1)}` });
+  data.append('name', name);
```

Comment:
Can you clean up some of the formatting from ln90 to 110?
Line breaks, indentation, etc